assert/unreachable.ts

```diff
@@ -11,6 +11,6 @@ import { AssertionError } from "./assertion_error.ts";
  * unreachable(); // Throws
  * ```
  */
-export function unreachable(): never {
-  throw new AssertionError("unreachable");
+export function unreachable(reason?: string): never {
```

Comment:
We usually use the `msg` argument name in `std/assert`. Though, how this is now might be fine.